src/views/Game/hooks/useProgressionListener.tsx

```diff
@@ -0,0 +1,79 @@
+import { useState, useEffect } from 'react';
+import { useSelector, useDispatch, batch } from 'react-redux';
+import { pipe } from 'fp-ts/lib/pipeable';
+import * as A from 'fp-ts/lib/Array';
+import * as O from 'fp-ts/lib/Option';
+
+import {
+  getCurrentPlay,
+  getCurrentPlayer,
+  getOrderedProgression,
+  getStartedSession,
+} from '../../../store/session/selectors';
+import { requestProgressionListener } from '../../../api/db/gameSession';
+import {
+  setCurrentPlay,
+  setCurrentPlayer,
+  setGameProgression,
+} from '../../../store/session/actions';
+import { isPass, PlayWithId } from '../../../model/Play';
+import { noop } from '../../../utils/unit';
+import { isOwnerOfPlay, getNextPlayer } from '../helpers/plays';
+import { getPlayerValue } from '../../../store/playerHand/selector';
+
+export function useProgressionListener() {
+  const player = useSelector(getPlayerValue);
+  const currentSession = useSelector(getStartedSession);
+  const currentPlay = useSelector(getCurrentPlay);
+  const currentPlayer = useSelector(getCurrentPlayer);
+  const orderedProgression = useSelector(getOrderedProgression);
+  const [hasListener, setHasListener] = useState<boolean>(false);
+  const dispatch = useDispatch();
+
+  function runNextEffect(play: PlayWithId) {
+    const nextPlayer = getNextPlayer(play, currentSession);
+
+    batch(() => {
+      dispatch(setCurrentPlay(play.id));
+      dispatch(setCurrentPlayer(nextPlayer.id));
```

Comment:
If we gonna have players making "out of their turn" plays, this might need some changes

Reply:
Or renaming for something more specific for a kind of play

Reply:
Not sure I get it, when the player makes out of turn plays we dispatch the same things.